### PR TITLE
Pin cryptography version

### LINKED ISSUE #195: 'cryptography' package requires a Rust compiler
We're getting an error in https://github.com/returntocorp/semgrep/pull/2515/checks?check_run_id=1851883158 indicating that installing the cryptography package requires a Rust compiler.

image

Announcement regarding dependency on rust: https://mail.python.org/pipermail/cryptography-dev/2020-December/000998.html
How to install rust: https://www.rust-lang.org/tools/install

### PR DESCRIPTION
**Pull request by nbrahms**
crypto 3.4+ depends on the Rust compiler, and crypto 3+ is required to
install poetry. Avoid this by pinning cryptography.

Fixes #195.

### CODE REVIEW

**Review by brendongo — APPROVED** [review]

File: Dockerfile
Comment (underyx) [inline Dockerfile]: We shouldn't split into two `RUN` instructions; that negates all our disk space saving measures right below this line. If the above `RUN` finishes before uninstalling stuff, then `poetry`, `libffi-dev`, etc. will all be committed to the final image. Compressed image size went from [86 MB](https://hub.docker.com/layers/returntocorp/semgrep-agent/v1/images/sha256-fd030566af43ead16e15c81740b38da076f0563ec8ae69ee70ed1fc8519b0e87?context=explore) to [206 MB](https://hub.docker.com/layers/returntocorp/semgrep-agent/pr-197/images/sha256-f2f02b2b14adb9440154e8b1cf59b2287e79d443598998f9f3e2b7190a84797c?context=explore).

**Review by underyx — DISMISSED** [review]
Please fix the image size. I'd fix myself but then I'd just be stuck waiting for an approval on my change anyway 😅

File: Dockerfile
Comment (underyx) [inline Dockerfile]:
```suggestion
    poetry install --no-dev --no-root &&\
    pip uninstall -y poetry &&\
```

**Review by underyx — APPROVED** [review]

**Pull request merged**
